teraserver/python/modules/FlaskModule/API/user/UserQueryDeviceTypes.py

```diff
@@ -0,0 +1,202 @@
+                print(sys.exc_info())
+                return gettext('Database error'), 500
+        else:
+            # New
+            try:
+                new_device_type = TeraDeviceType()
+                new_device_type.from_json(json_device_type)
+                TeraDeviceType.insert(new_device_type)
+                # Update ID for further use
+                json_device_type['id_device_type'] = new_device_type.id_device_type
+            except exc.SQLAlchemyError:
+                import sys
+                print(sys.exc_info())
+                return gettext('Database error'), 500
+
+        # TODO: Publish update to everyone who is subscribed to devices update...
+        update_device = TeraDeviceType.get_device_type_by_id(json_device_type['id_device_type'])
+
+        return [update_device.to_json()]
+
+    @user_multi_auth.login_required
+    @api.expect(delete_parser)
+    @api.doc(description='Delete a specific device type',
+             responses={200: 'Success',
+                        403: 'Logged user can\'t delete device type (can delete if site admin)',
+                        500: 'Device type not found or database error.',
+                        501: 'Tried to delete two devices at the same time (Name != ID)'})
+    def delete(self):
+        parser = delete_parser
+        current_user = TeraUser.get_user_by_uuid(session['_user_id'])
+        user_access = DBManager.userAccess(current_user)
+
+        args = parser.parse_args()
+        id_todel = args['id_device_type']
+        key_todel = args['device_type_key']
+
+        # Check if current user can delete
+        TeraDevice_todel_1 = TeraDeviceType.get_device_type_by_id(id_todel)
+        TeraDevice_todel_2 = TeraDeviceType.get_device_type_by_key(key_todel)
+
+        # If not argument or both argument incorrect
+        if TeraDevice_todel_1 is None and TeraDevice_todel_2 is None:
+            return gettext('Device type not found'), 500
+
+        # ID present and name is blank
+        if TeraDevice_todel_2 is None:
+            if TeraDevice_todel_1.id_device_type not in user_access.get_accessible_devices_types_ids(admin_only=True):
+                return gettext('Forbidden'), 403
+            try:
+                TeraDeviceType.delete(id_todel=TeraDevice_todel_1.id_device_type)
+            except exc.SQLAlchemyError:
+                import sys
+                print(sys.exc_info())
+                return gettext('Database error'), 500
+
+        # ID absent, name present
+        elif TeraDevice_todel_1 is None:
+            if TeraDevice_todel_2.id_device_type not in user_access.get_accessible_devices_types_ids(admin_only=True):
+                return gettext('Forbidden'), 403
+            try:
+                TeraDeviceType.delete(id_todel=TeraDevice_todel_2.id_device_type)
+            except exc.SQLAlchemyError:
+                import sys
+                print(sys.exc_info())
+                return gettext('Database error'), 500
```

Comment:
This could be simplified by querying a single object depending on the parameter:
if args['id_device_type']:
device_type_to_del = TeraDevice.get_device_type_by_id(args['id_device_type'])
elif args['device_type_key']:
device_type_to_del = TeraDevice.get_device_type_by_key(args['device_type_key'])
else:
return gettext('Missing arguments'), 400

Then, only the device_type_to_del  variable would have to be handled, whatever the arguments, removing duplicate codes.